lib/element/locate/elements-by-recursion.js

```diff
@@ -35,7 +35,23 @@ class MultipleElementsByRecursion extends RecursiveLookupBase {
         }
       });
 
-    return this.deferred.promise;
+    return this.deferred.promise
+      .then((result) => {
+        // backward compatibility: keep the "result" on the result object
```

Comment:
This change is required to maintain backward compatibility with `elements` calls as much as possible. 
Previously the `elements` function inside section commands returned only the first element in the `value` property. 

see this test: https://github.com/nightwatchjs/nightwatch/pull/2589/files#diff-638be4835910b0e93ad6f5357c40afcf51a73583209b49bb6ecf5602938bcd71L134

I agree this code is _aweful_ but I have not found any other way to actually return the correct value (array with elements: `[{ ELEMENT: '1' }, { ... }, ...]`) and maintain backward compatibility with respect the `result` property...